examples/container/memeater/src/main.rs

```diff
@@ -30,3 +32,24 @@ fn main() {
         println!("{}", x[0]);
     }
 }
+
+/// Returns the amount in bytes corresponding to all the memory pages maped by the process.
+fn get_used_memory_in_bytes() -> u64 {
+    let stat = fs::read_to_string("/proc/self/stat").expect("Could not read /proc/self/stat");
+    let mut fields = stat.split_whitespace();
+
+    // Read the Resident Set Size (RSS).
+    // This number correspond to the total number of memory pages used by the process.
+    // This includes also the shared libraries which means that the final count is larger
+    // than the real amount of bytes consumed by the process.
+    let rss = fields
+        .nth(23)
+        .expect("Could not read RSS memory usage")
+        .parse::<u64>()
+        .expect("Could not parse RSS");
+
+    // Usually the page size is set to 4kB by default
+    let page_size = 4096;
```

Comment:
Can we use `getpagesize()` for that?

Reply:
that would introduce a dependency to either libc or nix, which @flxo recommended to avoid to keep this example simple.

Reply:
you are right...skip it